### PR TITLE
Confirm match result

### LINKED ISSUE #6: Confirm/deny match result
In order to agree or not with a match result submitted by another user
As a player
I want to confirm or reject a match result

### PR DESCRIPTION
**Pull request by deinok**
Closes #6

### CODE REVIEW

**Review by bergacat1 — CHANGES_REQUESTED** [review]
First of all, in the scenarios of "MatchResultVerification" you say you are going to test as Player and as Admin but you login as Admin in all cases.

Secondly, you should test that the MatchResultVerification has been correctly linked with MatchResult and the Player. You should also do that for the tests of MatchResult, checking that the MatchResult is linked with the Players and Match.

Finally, you should check that a Player can only verify results of his matches.

**Comment by deinok** [comment]
@bergacat1 So how can we link a matchResult to a match, and all this things in gherkins?
Its okey as a toy tool but we waste more time testing that the frameworks we use are working as expected than implementing real features.

Should we implement a dictionary<string,object> that saves the results of one gherkins sentence to be able to get it in the next sentence?

**Comment by bergacat1** [comment]
The correct way would be to identify the match you want to verify by its characteristics (specially the date) so that the sentence would be reusable from anywhere. In your case, an example would be:
  ```
Scenario: A player creates a new MatchResultVerification
	Given I login as "testplayer@mypadel.cat" with password "password"
	And The player with email "testplayer@mypadel.cat" played a public match on "<date>"
	And There is a MatchResult for the public match player on "<date>" 
	When I agree through a MatchResultVerification with the MatchResult of the match played on "<date>"
	Then The response code is 201
	And A new MatchResultVerification is added for the MatchResult of the match played on "<date>"
```
By this way, you could also test that only players who played a certain match can submit or verify its result.

**Comment by deinok** [comment]
@bergacat1 Creiem que ja esta

File: src/test/java/cat/udl/eps/softarch/mypadel/steps/MatchResultVerificationCreateSteps.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	@And("^A new MatchResultVerification is added for the MatchResult of the match played on \"([^\"]*)\"$")
+	public void aNewMatchResultVerificationIsAddedForTheMatchResultOfTheMatchPlayedOn(String dateString) throws Throwable {
+		stepDefs.mockMvc.perform(
+			get("/matchResultVerifications/1")
+				.accept(MediaType.APPLICATION_JSON)
+				.with(authenticate()))
+			.andExpect(jsonPath("$.id", is(1)));
+	}
```
Comment (bergacat1) [inline src/test/java/cat/udl/eps/softarch/mypadel/steps/MatchResultVerificationCreateSteps.java]: Aquí haurieu de comprovar que la MatchResultVerification ha quedat correctament relacionada amb el MatchResult i el Player.

**Review by bergacat1 — CHANGES_REQUESTED** [review]
En l'últim 'step', us falta comprovar que el MatchResultVerification queda correctament relacionat amb el MatchResult i el Player.

**Comment by deinok** [comment]
@bergacat1 fet

**Review by bergacat1 — CHANGES_REQUESTED** [review]
El link estaria present encara que no hi hagués cap element associat. Heu de fer un GET a les URIs de les relacions i comprovar que retorna un 200. Si fossin null, retornaria 404.

**Comment by deinok** [comment]
@bergacat1 Fet.

PD: Si per cada propietat de navegacio ens hem de complicar tant, potse val la pena que implementem el protocol GraphQL http://graphql.org/

Al treball hem conseguit fer-ho amb navegacio infinita i cargan les entitats amb de forma "explicita" (sense el famos problema dels ORMs de "SELECT N+1").
Tambe ens simplificarie el treball al Frontend

**Review by bergacat1 — APPROVED** [review]
Ara perfecte. Molt interessant el GraphQL, però per aquest projecte millor no complicar-nos a aquestes altures...

**Pull request merged**